modules/features2d/src/sift.simd.hpp

```diff
@@ -829,10 +829,14 @@ else // CV_8U
         v_pack_store(dst + k, __pack01);
     }
 #endif
+// avoid warning "iteration 7 invokes undefined behavior" on Linux ARM64
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waggressive-loop-optimizations"
```

Comment:
Large code pieces should have `#if` condition with GCC version check and/or affected target.